### ISSUE TITLE
Memory leak when using slide transition on Android (API Level 23)

### ISSUE THREAD

**Issue by shripalsoni04** [NativeScript/NativeScript#2741]
### Did you verify this is a real problem by searching [Stack Overflow]
Yes

### Tell us about the problem
In my sample application to show the memory leak issue, I have two pages. On first page there is only button and on tap on that button, we are navigated to page 2. On Page 2, I have a Label in StackLayout.

Now, when we set page transition to 'slide', it is causing a memory leak on Android API Level 23. Memory keep on increasing on transition till OOM. When I tested that on emulator with API Level 21 and 22, It is not causing any memory leak.

Please check the below heap dump taken on API 23 version showing ImageViews being generated on slide transition are remaining in memory. 
![memory-leak-heap-dump](https://cloud.githubusercontent.com/assets/4411891/18598616/61193160-7c71-11e6-9ed6-81060572f5d9.png)

Attaching the path2gc for this image, which may help you in finding the memory leak root.
![path2gc](https://cloud.githubusercontent.com/assets/4411891/18598670/abea122c-7c71-11e6-80b7-fa02d453dd53.png)

When I checked, what these images are, I found that it is creating screenshot of both the pages on navigation. Below are that Images, which are keep on adding in memory on transition.
![image1](https://cloud.githubusercontent.com/assets/4411891/18598688/cceb5e9a-7c71-11e6-9be2-903f6f0cd047.png)

![image2](https://cloud.githubusercontent.com/assets/4411891/18598689/ccf155f2-7c71-11e6-9ca8-3113d5994969.png)

Also, I found that, it is not releasing Label in Page 2 on transition. So Labels also keep on adding into memory.

### Which platform(s) does your issue occur on?
Android

### Please provide the following version numbers that your issue occurs with:
- CLI: 2.3.0
- Cross-platform modules: Tested on both 2.3.0 and 2.4.0-2016-09-16-4127
- Runtime(s): tns-android: 2.3.0

### Please tell us how to recreate the issue in as much detail as possible.
1. Clone this [sample app](https://github.com/shripalsoni04/ns-transition-memory-leak)
2. Run on Android API 23.
3. Do transition between two pages for around 10 times.
4. You can either log heapMemory on button click, or take heapDump from Android Monitor and analyse it in Android Studio or Eclipse MAT.

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.

Basic sample app to reproduce this issue: https://github.com/shripalsoni04/ns-transition-memory-leak

**Comment by colinskow** [comment]
See the discussion at the bottom of #2571. I believe this is already fixed in master. Install `tns-core-modules@next` and test again.

**Comment by colinskow** [comment]
Never mind, it looks like you are using the latest release already. I have additional tests for this issue at https://github.com/colinskow/tns-image-test

**Comment by shripalsoni04** [comment]
@colinskow Thanks for sharing image tests. 

Actually in my case, the issue happens only when we use "Slide" transition for navigation on Android. It will work fine if you use "Fade" transition. In my sample app I don't have any image.  The memory leak happens event when we have only Label in the page.

**Comment by tsonevn** [comment]
Hi @shripalsoni04,

I tested the given scenario and I was able to reproduce this behavior with `Slider` transition while navigating and in non-NativeScript Angular2 project. I am attaching [here ](https://github.com/tsonevn/NavigateTransition_isuue2741.git) also my sample project, where this problem has been reproduced.


Regards,
@tsonevn

**Comment by hshristov** [comment]
Fixed with @next version of `tns-core-modules` and `tns-core-modules-widgets`. Could you please check with them if everything is ok?

**Comment by shripalsoni04** [comment]
Hi @hshristov,

When I tested this on genymotion emulator  (heapSize: 96MB, API: 23), the transition images were not getting cleared till 14 transitions and after that the GC got called and it cleared some memory. After that I continued to do the transitions for around 16 more time, to check it GC is getting called or not..But it didn't call and it threw OOM and app crashed. 

When I tested this on device (heapSize: 256MB, API: 23), the transition images were kept on adding on memory till around 165 MB and then GC called and cleared some memory. I keep on doing transitions, and again around 165MB the GC called and cleared some memory. But it is not clearing all the memory. I analysed the heap dump and found that still some transition images are not getting removed from memory.

So I think the issue is still there. Attaching the path2gc for one of the transition image, which may help you for further investigation.

![screen shot 2016-10-15 at 2 00 34 pm](https://cloud.githubusercontent.com/assets/4411891/19408641/1d2bba74-92e0-11e6-9390-f9a71bacf955.png)

Let me know if you want any other information.
